source/priority_queue.c

```diff
@@ -44,15 +48,20 @@ static void s_swap(struct aws_priority_queue *queue, size_t a, size_t b) {
             (*bp_b)->current_index = b;
         }
     }
+    AWS_POSTCONDITION(aws_priority_queue_is_valid(queue));
 }
 
 /* Precondition: with the exception of the given root element, the container must be
  * in heap order */
 static bool s_sift_down(struct aws_priority_queue *queue, size_t root) {
+    AWS_PRECONDITION(aws_priority_queue_is_valid(queue));
+    AWS_PRECONDITION(root < queue->container.length);
+
     bool did_move = false;
 
     size_t len = aws_array_list_length(&queue->container);
 
+    /* Note: There is an unwinding loop assertion here. */
```

Comment:
I'm not sure what to make of this comment? In particular as it's part of the source code?

Reply:
I forgot to remove that, thanks for noticing.